app/src/main/java/mozilla/lockbox/presenter/RoutePresenter.kt

```diff
@@ -54,12 +59,39 @@ class RoutePresenter(
             is RouteAction.OpenWebsite -> {
                 openWebsite(destination.url)
             }
+            is RouteAction.SystemSetting -> {
+                openSetting(destination.setting.settingIntent)
+            }
             is RouteAction.FingerprintDialog -> showDialogFragment(FingerprintAuthDialogFragment())
+            is RouteAction.DialogAction -> showDialog(destination)
 
             is RouteAction.Back -> navController.popBackStack()
         }
     }
 
+    private fun showDialog(destination: RouteAction.DialogAction) {
+        val dialogStateObservable = when (destination) {
+            is RouteAction.DialogAction.SecurityDisclaimerDialog -> showSecurityDisclaimerDialog()
+        }
+
+        dialogStateObservable
+            .subscribe { alertState ->
+                val action = when (alertState) {
+                    AlertState.BUTTON_POSITIVE -> {
+                        destination.positiveButtonAction
+                    }
+                    AlertState.BUTTON_NEGATIVE -> {
+                        destination.negativeButtonAction
+                    }
+                }
+
+                action?.let {
+                    dispatcher.dispatch(action)
+                }
+            }
+            .addTo(compositeDisposable)
```

Comment:
Presenter. 

I'm still a little bit icky about having this much code in the `RoutePresenter`.  I'm happy to land this, and wait for it to be more apparent we need to refactor it (it may not).